BUILD.gn

```diff
@@ -213,6 +213,7 @@ if (current_toolchain != "${dir_pw_toolchain}/default:default") {
       deps += [
         "${chip_root}/src/app/server/java",
         "${chip_root}/src/controller/java",
+        "${chip_root}/src/controller/tests/java",
```

Comment:
What will this test?

Trying to figure out if a real controller java test (in which case location is correct) or an app that would exercise the controller java api (in which case this is an 'example app' and it should go in examples).

I think this may be the latter, so path should be in examples and should not be built by default.

Reply:
good question, I think currently we are targeting light-weight controller testing include commissioning and IM interaction? Are we expecting more beyond controller part?

Reply:
@mrjerryjohns @yufengwangca  thoughts?

Reply:
I think finally we should use java test app to cover all functionality in sdk? so maybe it is better to go with yufeng's proposal with new x86 example app?